### DISCUSSION TITLE
image:exif:DateTimeOriginal timezone

### DISCUSSION THREAD

**Discussion by gisms** [sepinf-inc/IPED#1215] [Q&A]
Hi.

I am checking the metadata from em jpg file and IPED gave the image:exif:DateTimeOriginal = 15:57:34Z (zulu timezone or UTC).
When I used exif tool, the result was: Date/Time Original: 15:57:34.992-03:00.
The same hour, but indicanting the correct timezone.

Is this divergence already known?

**Comment by lfcnassif** [comment]
It's not. Could you share the file (privately if needed), so we could take a closer look?

**Reply by gisms** [reply]
Yes, I managed to apply the behavior on a test image, i will send to you.

**Reply by gisms** [reply]
image sent

**Reply by lfcnassif** [reply]
Thank you @gisms! I'll take a look soon.

**Reply by lfcnassif** [reply]
What iped version did you use?

**Reply by gisms** [reply]
4.0.0 beta and also I tested with 18.13.
When I eport the exif tool result (all tags) to a txt file, the timzeone information from, for example, -03:00 (Date/Time Original: 15:57:34.992-03:00.) is not include.
It is included when i execute from cmd.

**Reply by gisms** [reply]
In fact, exiftool is giving two information for **Date/Time Original**, in the first, he does not show the timezone. in the second, he does and keep the same value. the timezone -3 exists in the file:

File Size                       : 4.6 MiB
File Modification Date/Time     : 2022:06:16 11:18:50-03:00
File Access Date/Time           : 2022:07:11 17:43:25-03:00
File Creation Date/Time         : 2022:07:11 14:11:42-03:00
File Permissions                : -rw-rw-rw-
File Type                       : JPEG
File Type Extension             : jpg
MIME Type                       : image/jpeg
Exif Byte Order                 : Big-endian (Motorola, MM)
Copyright                       :
Resolution Unit                 : inches
Image Description               :
Make                            : motorola
Camera Model Name               : moto g(20)
Software                        :
Orientation                     : Unknown (0)
Modify Date                     : 2022:06:16 11:18:49
Y Cb Cr Positioning             : Centered
ISO                             : 50
Image Unique ID                 : IMAGE 2022:06:16 11:18:49
F Number                        : 1.7
Exposure Time                   : 1/410
Sub Sec Time Digitized          : 20
Offset Time Digitized           : -03:00
Sub Sec Time Original           : 100
Offset Time Original            : -03:00
Sub Sec Time                    : 10
Offset Time                     : -03:00
Focal Length                    : 4.7 mm
Flash                           : No Flash
Light Source                    : Other
Scene Capture Type              : Standard
Max Aperture Value              : 2.6
Create Date                     : 2022:06:16 11:18:49
Exif Image Height               : 3000
White Balance                   : Auto
**Date/Time Original**              : 2022:06:16 11:18:49
Exif Image Width                : 4000
Aperture Value                  : 2.6
Components Configuration        : Y, Cb, Cr, -
Color Space                     : sRGB
Exif Version                    : 0220
Flashpix Version                : 0100
GPS Img Direction Ref           : Magnetic North
GPS Latitude Ref                : South
GPS Img Direction               : 0
GPS Longitude Ref               : West
GPS Altitude Ref                : Above Sea Level
GPS Time Stamp                  : 14:18:47
GPS Processing Method           :
GPS Date Stamp                  : 2022:06:16
X Resolution                    : 72
Artist                          :
Y Resolution                    : 72
Thumbnail Offset                : 1135
Thumbnail Length                : 14636
Compression                     : JPEG (old-style)
XMP Toolkit                     : Adobe XMP Core 5.1.0-jc003
Special Type ID                 : HDR_PHOTO_TYPE
JFIF Version                    : 1.01
Image Width                     : 3000
Image Height                    : 4000
Encoding Process                : Baseline DCT, Huffman coding
Bits Per Sample                 : 8
Color Components                : 3
Y Cb Cr Sub Sampling            : YCbCr4:2:0 (2 2)
Aperture                        : 1.7
Image Size                      : 3000x4000
Megapixels                      : 12.0
Shutter Speed                   : 1/410
Create Date                     : 2022:06:16 11:18:49.20-03:00
**Date/Time Original**              : 2022:06:16 11:18:49.100-03:00
Modify Date                     : 2022:06:16 11:18:49.10-03:00
Thumbnail Image                 : (Binary data 14636 bytes, use -b option to extract)
GPS Altitude                    : 0 m Above Sea Level
GPS Date/Time                   : 2022:06:16 14:18:47Z
GPS Latitude                    : 16 deg 40' 52.92" S
GPS Longitude                   : 49 deg 15' 8.31" W
Focal Length                    : 4.7 mm
GPS Position                    : 16 deg 40' 52.92" S, 49 deg 15' 8.31" W
Light Value                     : 11.2

**Reply by lfcnassif** [reply]
Thanks @gisms. I just opened https://github.com/sepinf-inc/IPED/issues/1217. This seems a Tika library issue affecting us. Could I use the picture you sent to me to report the problem to them in their public bug tracker?

**Reply by gisms** [reply]
nassif, running the exiftool command with G1 option, to be more clear, we have [ExifIFD] Date/Time Original without the timezone, but the composite Date/Time Original apply the timezone and keep the same value. We dont have this composite field in any metadata field in iped.

you can use my photo, is personal.

[ExifTool]      ExifTool Version Number         : 12.29
[System]        File Name                       : IMG_20220616_111848_HDR.jpg
[System]        Directory                       : C:/Users/gisele.souza/Desktop/teste/IMG_20220616_111848_HDR
[System]        File Size                       : 4.6 MiB
[System]        File Modification Date/Time     : 2022:06:16 11:18:50-03:00
[System]        File Access Date/Time           : 2022:07:11 18:20:37-03:00
[System]        File Creation Date/Time         : 2022:07:11 14:11:42-03:00
[System]        File Permissions                : -rw-rw-rw-
[File]          File Type                       : JPEG
[File]          File Type Extension             : jpg
[File]          MIME Type                       : image/jpeg
[File]          Exif Byte Order                 : Big-endian (Motorola, MM)
[File]          Image Width                     : 3000
[File]          Image Height                    : 4000
[File]          Encoding Process                : Baseline DCT, Huffman coding
[File]          Bits Per Sample                 : 8
[File]          Color Components                : 3
[File]          Y Cb Cr Sub Sampling            : YCbCr4:2:0 (2 2)
[IFD0]          Image Width                     : 3000
[IFD0]          Image Height                    : 4000
[IFD0]          Copyright                       :
[IFD0]          Resolution Unit                 : inches
[IFD0]          Image Description               :
[IFD0]          Make                            : motorola
[IFD0]          Camera Model Name               : moto g(20)
[IFD0]          Software                        :
[IFD0]          Orientation                     : Unknown (0)
[IFD0]          Modify Date                     : 2022:06:16 11:18:49
[IFD0]          Y Cb Cr Positioning             : Centered
[IFD0]          X Resolution                    : 72
[IFD0]          Artist                          :
[IFD0]          Y Resolution                    : 72
[ExifIFD]       ISO                             : 50
[ExifIFD]       Image Unique ID                 : IMAGE 2022:06:16 11:18:49
[ExifIFD]       F Number                        : 1.7
[ExifIFD]       Exposure Time                   : 1/410
[ExifIFD]       Sub Sec Time Digitized          : 20
[ExifIFD]       Offset Time Digitized           : -03:00
[ExifIFD]       Sub Sec Time Original           : 100
[ExifIFD]       Offset Time Original            : -03:00
[ExifIFD]       Sub Sec Time                    : 10
[ExifIFD]       Offset Time                     : -03:00
[ExifIFD]       Focal Length                    : 4.7 mm
[ExifIFD]       Flash                           : No Flash
[ExifIFD]       Light Source                    : Other
[ExifIFD]       Scene Capture Type              : Standard
[ExifIFD]       Max Aperture Value              : 2.6
[ExifIFD]       Create Date                     : 2022:06:16 11:18:49
[ExifIFD]       Exif Image Height               : 3000
[ExifIFD]       White Balance                   : Auto
[ExifIFD]       Date/Time Original              : 2022:06:16 11:18:49
[ExifIFD]       Exif Image Width                : 4000
[ExifIFD]       Aperture Value                  : 2.6
[ExifIFD]       Components Configuration        : Y, Cb, Cr, -
[ExifIFD]       Color Space                     : sRGB
[ExifIFD]       Exif Version                    : 0220
[ExifIFD]       Flashpix Version                : 0100
[GPS]           GPS Img Direction Ref           : Magnetic North
[GPS]           GPS Latitude Ref                : South
[GPS]           GPS Img Direction               : 0
[GPS]           GPS Latitude                    : 16 deg 40' 52.92"
[GPS]           GPS Longitude Ref               : West
[GPS]           GPS Longitude                   : 49 deg 15' 8.31"
[GPS]           GPS Altitude Ref                : Above Sea Level
[GPS]           GPS Altitude                    : 0 m
[GPS]           GPS Time Stamp                  : 14:18:47
[GPS]           GPS Processing Method           :
[GPS]           GPS Date Stamp                  : 2022:06:16
[IFD1]          Image Width                     : 0
[IFD1]          Thumbnail Offset                : 1135
[IFD1]          Image Height                    : 0
[IFD1]          Orientation                     : Horizontal (normal)
[IFD1]          Thumbnail Length                : 14636
[IFD1]          Compression                     : JPEG (old-style)
[IFD1]          Y Cb Cr Positioning             : Centered
[IFD1]          Resolution Unit                 : inches
[IFD1]          X Resolution                    : 72
[IFD1]          Y Resolution                    : 72
[IFD1]          Thumbnail Image                 : (Binary data 14636 bytes, use -b option to extract)
[XMP-x]         XMP Toolkit                     : Adobe XMP Core 5.1.0-jc003
[XMP-GCamera]   Special Type ID                 : HDR_PHOTO_TYPE
[JFIF]          JFIF Version                    : 1.01
[JFIF]          Resolution Unit                 : None
[JFIF]          X Resolution                    : 1
[JFIF]          Y Resolution                    : 1
[Composite]     Aperture                        : 1.7
[Composite]     Image Size                      : 3000x4000
[Composite]     Megapixels                      : 12.0
[Composite]     Shutter Speed                   : 1/410
[Composite]     Create Date                     : 2022:06:16 11:18:49.20-03:00
[Composite]     Date/Time Original              : 2022:06:16 11:18:49.100-03:00
[Composite]     Modify Date                     : 2022:06:16 11:18:49.10-03:00
[Composite]     GPS Altitude                    : 0 m Above Sea Level
[Composite]     GPS Date/Time                   : 2022:06:16 14:18:47Z
[Composite]     GPS Latitude                    : 16 deg 40' 52.92" S
[Composite]     GPS Longitude                   : 49 deg 15' 8.31" W
[Composite]     Focal Length                    : 4.7 mm
[Composite]     GPS Position                    : 16 deg 40' 52.92" S, 49 deg 15' 8.31" W
[Composite]     Light Value                     : 11.2

**Reply by lfcnassif** [reply]
I think those exiftool [Composite] properties are combinations of other properties done by exiftool and are not stored directly by the file format, so we possibly aren't loosing stored metadata. But we are displaying inconsistent Date information indeed and this should be fixed.

**Reply by lfcnassif** [reply]
And, I'm not sure if the EXIF specification defines stored dates should always have timezone information or if some default timezone should be used if it is not specified, **maybe** it can change depending on camera model... So, **maybe**, some cases would be hard to choose the right value... We might leave dates without timezone when displaying them, but the timezone is really needed for the timeline, for date sorting and to do date range queries...

**Reply by lfcnassif** [reply]
I just found the bug in upstream library:
https://issues.apache.org/jira/browse/TIKA-3815?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&focusedCommentId=17565200#comment-17565200

**Reply by lfcnassif** [reply]
And I proposed a fix: https://github.com/apache/tika/pull/605

**Reply by lfcnassif** [reply]
> the EXIF specification

I read it: just `GPS Date/Time Stamp` includes timezone information by the specification.

**Reply by gisms** [reply]
> And, I'm not sure if the EXIF specification defines stored dates should always have timezone information or if some default timezone should be used if it is not specified, **maybe** it can change depending on camera model... So, **maybe**, some cases would be hard to choose the right value... We might leave dates without timezone when displaying them, but the timezone is really needed for the timeline, for date sorting and to do date range queries...

I agree with you.

**Reply by gisms** [reply]
> > the EXIF specification
> 
> I read it: just `GPS Date/Time Stamp` includes timezone information by the specification.

can you send me the link to this specificatin file?

**Reply by lfcnassif** [reply]
It's here http://web.archive.org/web/20131019050323/http://www.exif.org/specifications.html
